File: sdks/python/blocks_network/protocol_version.py
"""
Protocol versioning constants and helpers for the Blocks Python SDK.

This is the single source of truth for protocol version values in the
Python SDK. All modules that need version strings import from here.
"""

from __future__ import annotations

import importlib.metadata
from typing import List

# ---------------------------------------------------------------------------
# Protocol version constants
# ---------------------------------------------------------------------------

CURRENT_PROTOCOL_VERSION: str = "2026-05-01"
"""The current (preferred) Blocks wire protocol version."""

SUPPORTED_PROTOCOL_VERSIONS: List[str] = ["2026-05-01"]
"""All protocol versions this SDK can speak."""

DEPRECATED_PROTOCOL_VERSIONS: List[str] = []
"""Protocol versions that are still supported but deprecated."""

PROTOCOL_VERSION_HEADER: str = "Blocks-Protocol-Version"
"""HTTP header name for protocol version on all guarded endpoints."""

# ---------------------------------------------------------------------------
# SDK package version
# ---------------------------------------------------------------------------


def _resolve_sdk_version() -> str:
    """Resolve the SDK package version from installed metadata.

    Falls back to ``"unknown"`` if the package is not installed or
    metadata is unavailable.
    """
    try:
        return importlib.metadata.version("blocks-network")
    except Exception:
        return "unknown"


SDK_VERSION: str = _resolve_sdk_version()
"""The installed ``blocks-network`` Python package version."""

# ---------------------------------------------------------------------------
# Helpers
# ---------------------------------------------------------------------------


def is_supported(version: str) -> bool:
    """Return ``True`` if *version* is in the supported set."""
    return version in SUPPORTED_PROTOCOL_VERSIONS


def is_deprecated(version: str) -> bool:
    """Return ``True`` if *version* is deprecated but still supported."""
    return version in DEPRECATED_PROTOCOL_VERSIONS

File: sdks/node/src/runtime/protocol-version.ts
/**
 * Protocol versioning constants and helpers for the Blocks Node SDK.
 *
 * Centralizes the protocol version contract so no other file needs to
 * hardcode version strings or header names.
 */

/** Current protocol version used by this SDK build. */
export const CURRENT_PROTOCOL_VERSION = '2026-05-01';

/** All protocol versions this SDK can speak. */
export const SUPPORTED_PROTOCOL_VERSIONS: readonly string[] = [CURRENT_PROTOCOL_VERSION];

/** Protocol versions that are deprecated but still functional. */
export const DEPRECATED_PROTOCOL_VERSIONS: readonly string[] = [];

/** HTTP header name for protocol version negotiation. */
export const PROTOCOL_VERSION_HEADER = 'Blocks-Protocol-Version';

/** SDK package version (kept in sync with package.json manually). */
export const SDK_VERSION = '0.1.27';

/** Check whether a given protocol version is supported by this SDK. */
export function isProtocolVersionSupported(version: string): boolean {
  return SUPPORTED_PROTOCOL_VERSIONS.includes(version);
}
